lib/api/objectGet.js

```diff
@@ -106,6 +106,28 @@ function objectGet(authInfo, request, returnTagCount, log, callback) {
             // objMD.location is just a string
             dataLocator = Array.isArray(objMD.location) ?
                 objMD.location : [{ key: objMD.location }];
+
+            let partNumber = null;
+            if (request.query && request.query.partNumber !== undefined) {
+                if (byteRange) {
+                    const error = errors.InvalidRequest
+                        .customizeDescription('Cannot specify both Range ' +
+                            'header and partNumber query parameter.');
+                    return callback(error, null, corsHeaders);
+                }
+                partNumber = Number(request.query.partNumber, 10);
+                if (Number.isNaN(partNumber)) {
```

Comment:
I have added this check, even though the AWS SDK checks that the `PartNumber` param is a `Number`.